rust-toolchain

```diff
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "nightly-2023-02-03"
+channel = "nightly-2023-07-01"
```

Comment:
in my previous merge, I upgrade the toolchain version to 1.71.1 because one of the dependency requires the 1.70+ version. not sure whether stable version is better or stick with nightly version is better

Reply:
Frankly, stable should be better if all the functions we need are satisfied. We can change to stable